### PR TITLE
:sparkles: Controllable component

### PR DESCRIPTION
**Pull request by Tugduoff**
Added new Controllable component.

This component takes a map of actions mapped to keys.
Both actions and keys enum are included in this pull request.

It can be created through a default ctor, a ctor that takes an unordered map of keyMaps or a ctor that takes a libconfig::Setting.

It can be added dynamically through addTo functions override.

An example of this dynamic adding is made in the config/config.cfg file where we specify what actions and keys we use for the component.

If you have any questions, feel free to ask !

### CODE REVIEW

**Review by tibaudlemaout — APPROVED** [review]
It all seems good to me :+1:

**Review by C0Florent — APPROVED** [review]
Seems fine

File: Keys.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#include "Keys.hpp"
+
+Key stringToKey(const std::string &keyString) {
+    auto it = strToKey.find(keyString);
+    if (it != strToKey.end()) {
+        return it->second;
+    } else {
+        return Key::UNKNOWN;
+    }
+}
```
[the remark this answers is not in the document]
Comment (C0Florent) [reply Keys.cpp:17]: My bad, I forgot Key is an enum, not a class and you can't define constructors for enums

File: Actions.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Action stringToAction(const std::string &actionString) {
+    auto it = strToAction.find(actionString);
+    if (it != strToAction.end()) {
+        return it->second;
+    } else {
+        return Action::UNKNOWN;
+    }
+}
+
```
[the remark this answers is not in the document]
Comment (C0Florent) [reply Actions.cpp:18]: My bad, I forgot Action is an enum, not a class and you can't define constructors for enums

**Pull request merged**